proxy/handlers/handlers.go

```diff
@@ -170,27 +170,11 @@ tagLoop:
 		}
 	}()
 
-	errorChannel := make(chan error)
-	select {
-	case destination.SendChannel() <- connect.SendRequest{
-		Metric:       metric,
-		ErrorChannel: errorChannel,
-	}:
-		err = <-errorChannel
-		if err != nil {
-			proxy.Statsd.Count(
-				"veneur_proxy.handle.metrics_count",
-				int64(1), []string{"error:forward"}, 1.0)
-		} else {
-			proxy.Statsd.Count(
-				"veneur_proxy.handle.metrics_count",
-				int64(1), []string{"error:false"}, 1.0)
-		}
-		return
-	default:
-		proxy.Statsd.Count(
-			"veneur_proxy.handle.metrics_count",
-			int64(1), []string{"error:enqueue"}, 1.0)
-		return
+	destination.SendChannel() <- connect.SendRequest{
```

Comment:
As discussed offline I suspect this will improve throughput until the channel fills, then revert to the current behaviour (accidental synchronization with the destination worker).

Since the sendbuffer sizes are relatively arbitrary right now, it may be useful to measure. I would be interested in preserving that `error:enqueue` metric. One way to do so might be to attempt a nonblocking send, then emit+attempt a blocking send if it fails.

Reply:
I would also recommend adding some commentary here, since it's performance sensitive and under scrutiny. It is not immediately obvious that there's buffering, when we'd expect this handler to block, what happens if it does, etc.

Reply:
Done, thanks!